e2etest/bgptests/exclude_node.go

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package bgptests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.universe.tf/e2etest/pkg/k8s"
+	"go.universe.tf/e2etest/pkg/k8sclient"
+	"go.universe.tf/e2etest/pkg/metallb"
+
+	frrconfig "go.universe.tf/e2etest/pkg/frr/config"
+	"go.universe.tf/e2etest/pkg/ipfamily"
+	testservice "go.universe.tf/e2etest/pkg/service"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+var _ = ginkgo.Describe("FRR BGP ExcludeNode IPV4", func() {
+	var (
+		ann           = [2]string{"ocp-mc", "rebooting"}
+		allNodes      *corev1.NodeList
+		excludeCM     = "excludenodepattern" // configmap name is hardcoded to manifests
+		cs            clientset.Interface
+		svc           *corev1.Service
+		testNamespace string
+	)
+
+	restartSpeakerPods := func() {
+		err := metallb.RestartSpeakerPods(cs)
+		Expect(err).NotTo(HaveOccurred())
+		Eventually(func() error {
+			pods, err := metallb.SpeakerPods(cs)
+			if err != nil {
+				return err
+			}
+
+			for _, p := range pods {
+				if !k8s.PodIsReady(p) {
+					return fmt.Errorf("speaker pods are not ready")
+				}
+			}
+
+			return nil
+		}, 3*time.Minute, 5*time.Second).ShouldNot(HaveOccurred())
+
+		err = FRRProvider.Update()
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	ginkgo.BeforeEach(func() {
+		err := ConfigUpdater.Clean()
+		Expect(err).NotTo(HaveOccurred(), "cleaning k8s api CRs failed")
+
+		data := `metadata:
+      annotations:
+        %s: %s
+      labels:
+        foo: "bar"`
+
+		cm := map[string]string{"pattern.yaml": fmt.Sprintf(data, ann[0], ann[1])}
+
+		cs = k8sclient.New()
+		ginkgo.By("adding the excludenodepattern configmap")
+		err = k8s.CreateConfigmap(cs, excludeCM, metallb.Namespace, cm)
+		Expect(err).NotTo(HaveOccurred())
+
+		ginkgo.By("restarting speaker pods so excludenode configmap is read")
+		restartSpeakerPods()
+
+		for _, c := range FRRContainers {
+			err := c.UpdateBGPConfigFile(frrconfig.Empty)
+			Expect(err).NotTo(HaveOccurred(),
+				fmt.Sprintf("cleaning frr config at %s failed", c.Name))
+		}
+
+		testNamespace, err = k8s.CreateTestNamespace(cs, "bgp")
+		Expect(err).NotTo(HaveOccurred())
+
+		allNodes, err = cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		_, svc = setupBGPService(cs, testNamespace, ipfamily.IPv4, []string{v4PoolAddresses}, FRRContainers, func(svc *corev1.Service) {
+			testservice.TrafficPolicyCluster(svc)
+		})
+		testservice.ValidateDesiredLB(svc)
```

Comment:
what's the purpouse of validating this in BeforeEach?

Reply:
removed